### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by mnonnenmacher**
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/145)

### CODE REVIEW

File: analyzer/src/funTest/kotlin/integration/AbstractIntegrationSpec.kt [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+import io.kotlintest.matchers.beEmpty
+import io.kotlintest.matchers.shouldBe
+import io.kotlintest.matchers.shouldNot
+import io.kotlintest.matchers.shouldNotBe
+import io.kotlintest.specs.StringSpec
+
+import java.io.File
+
+abstract class AbstractIntegrationSpec : StringSpec() {
+
```
Comment (sschuberth) [inline analyzer/src/funTest/kotlin/integration/AbstractIntegrationSpec.kt]: Nit: I thought you adapted to my style ;-)

File: analyzer/src/funTest/kotlin/integration/AbstractIntegrationSpec.kt [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            downloadDir shouldNotBe null
+            workingDir shouldNotBe null
+            workingDir!!.isValid() shouldBe true
+            workingDir.getProvider() shouldBe pkg.vcs.provider
+        }.config(tags = setOf(Expensive))
+
+        "All package manager definition files are found" {
+            val definitionFiles = PackageManager.findManagedFiles(downloadDir)
+
+            definitionFiles.size shouldBe expectedDefinitionFiles.size
```
Comment (sschuberth) [inline analyzer/src/funTest/kotlin/integration/AbstractIntegrationSpec.kt]: Shouldn't `expectedDefinitionFiles.size` be `definitionFilesForTest.size`?

File: analyzer/src/funTest/kotlin/integration/AbstractIntegrationSpec.kt [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                results.values.forEach { result ->
+                    result.project.vcs.provider shouldBe pkg.vcs.provider
+                    result.project.vcs.url shouldBe pkg.vcs.url
+                    result.project.scopes shouldNot beEmpty()
+                    result.packages shouldNot beEmpty()
+                }
+            }
+        }.config(tags = setOf(Expensive))
+    }
+
```
Comment (sschuberth) [inline analyzer/src/funTest/kotlin/integration/AbstractIntegrationSpec.kt]: Nit: I thought you adapted to my style 😉

File: analyzer/src/funTest/kotlin/integration/GradleIntegrationTest.kt [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+import com.here.ort.analyzer.managers.Gradle
+import com.here.ort.analyzer.managers.Maven
+import com.here.ort.model.Package
+import com.here.ort.model.RemoteArtifact
+import com.here.ort.model.VcsInfo
+
+import java.io.File
+
+class GradleIntegrationTest : AbstractIntegrationSpec() {
+
```
Comment (sschuberth) [inline analyzer/src/funTest/kotlin/integration/GradleIntegrationTest.kt]: Nit: I thought you adapted to my style 😉

File: analyzer/src/funTest/kotlin/integration/GradleIntegrationTest.kt [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        // The Gradle project contains far too many definition files to list them all here. Use this tests to double
+        // check that all of them are found, and that they are assigned to the correct package manager.
+        val gradleFilenames = listOf("build.gradle", "settings.gradle")
+        val gradleFiles = downloadDir.walkTopDown().filter { it.name in gradleFilenames }.toMutableList()
+
+        // settings.gradle shall only be detected if there is no build.gradle file in the same directory.
+        val duplicates = gradleFiles.filter {
+            it.name == "settings.gradle" && gradleFiles.contains(File(it.parent, "build.gradle"))
+        }
+        gradleFiles.removeAll(duplicates)
```
Comment (sschuberth) [inline analyzer/src/funTest/kotlin/integration/GradleIntegrationTest.kt]: You could use `removeAll {  }` right away.

File: analyzer/src/funTest/kotlin/integration/GradleIntegrationTest.kt [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        mapOf(
+                Gradle to gradleFiles,
+                Maven to downloadDir.walkTopDown().filter { it.name == "pom.xml" }.toList()
+        )
+    }
+
+    override val definitionFilesForTest by lazy {
+        mapOf(Gradle as PackageManagerFactory<PackageManager> to listOf(File(downloadDir, "buildSrc/build.gradle")))
+    }
+
```
Comment (sschuberth) [inline analyzer/src/funTest/kotlin/integration/GradleIntegrationTest.kt]: Nit: I thought you adapted to my style 😉

File: analyzer/src/funTest/kotlin/integration/OssReviewToolkitIntegrationTest.kt [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+import com.here.ort.analyzer.managers.PIP
+import com.here.ort.analyzer.managers.SBT
+import com.here.ort.model.Package
+import com.here.ort.model.RemoteArtifact
+import com.here.ort.model.VcsInfo
+
+import java.io.File
+
+class OssReviewToolkitIntegrationTest : AbstractIntegrationSpec() {
+
```
Comment (sschuberth) [inline analyzer/src/funTest/kotlin/integration/OssReviewToolkitIntegrationTest.kt]: Nit: I thought you adapted to my style 😉

File: analyzer/src/funTest/kotlin/integration/OssReviewToolkitIntegrationTest.kt [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+                SBT to listOf(
+                        File(downloadDir, "analyzer/src/funTest/assets/projects/synthetic/all-managers/build.sbt")
+                )
+        )
+    }
+
+    override val definitionFilesForTest by lazy {
+        mapOf(Gradle as PackageManagerFactory<PackageManager> to listOf(File(downloadDir, "analyzer/build.gradle")))
+    }
+
```
Comment (sschuberth) [inline analyzer/src/funTest/kotlin/integration/OssReviewToolkitIntegrationTest.kt]: Nit: I thought you adapted to my style 😉

File: analyzer/src/funTest/kotlin/integration/AbstractIntegrationSpec.kt (hunk 1 above)
Comment (mnonnenmacher) [reply analyzer/src/funTest/kotlin/integration/AbstractIntegrationSpec.kt]: Only for functions but not for classes, it seems ;)

File: analyzer/src/funTest/kotlin/integration/AbstractIntegrationSpec.kt (hunk 2 above)
Comment (mnonnenmacher) [reply analyzer/src/funTest/kotlin/integration/AbstractIntegrationSpec.kt]: No, expectedDefinitionFiles is for this test, while definitionFilesForTest is for the dependency resolution test below.

File: analyzer/src/funTest/kotlin/integration/OssReviewToolkitIntegrationTest.kt (hunk 8 above)
Comment (mnonnenmacher) [reply analyzer/src/funTest/kotlin/integration/OssReviewToolkitIntegrationTest.kt]: I don't really mind, do you want me to change it?

Comment (sschuberth) [reply analyzer/src/funTest/kotlin/integration/OssReviewToolkitIntegrationTest.kt]: It's not necessary, no.

Comment (mnonnenmacher) [reply analyzer/src/funTest/kotlin/integration/OssReviewToolkitIntegrationTest.kt]: I missed this one, done.

**Pull request merged**